source/http_filesystem/http_filesystem.go

```diff
@@ -0,0 +1,128 @@
+package httpfs
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path"
+	"sort"
+
+	"github.com/golang-migrate/migrate/v4/source"
+)
+
+func init() {
+	source.Register("http-fs", &httpFs{})
```

Comment:
Don't register the source since you can't use `source.Open()` with this "driver" anyways.

Reply:
Done.